localstack/services/cloudformation/service_models.py

```diff
@@ -775,20 +775,25 @@ def cloudformation_type():
 
     def fetch_state(self, stack_name, resources):
         props = self.props
-        api_id = self.resolve_refs_recursively(stack_name, props['RestApiId'], resources)
-        res_id = self.resolve_refs_recursively(stack_name, props['ResourceId'], resources)
+        try:
+            api_id = self.resolve_refs_recursively(stack_name, props['RestApiId'], resources)
+            res_id = self.resolve_refs_recursively(stack_name, props['ResourceId'], resources)
+            if not api_id or not res_id:
+                return None
 
-        if not api_id or not res_id:
+        except DependencyNotYetSatisfied:
```

Comment:
Thanks for this PR @duongpv7 . We shouldn't have to handle this type of exception here in the model classes - this situation should be handled in `template_deployer.py` or `cloudformation_api.py`, depending on where the call originates from. Do you have a stack trace you can share, so we can take a closer look? Thanks!

Reply:
Hi @whummer 
> res_id = self.resolve_refs_recursively(stack_name, props['ResourceId'], resources)

This can raise an exception `DependencyNotYetSatisfied` because it can be called before dependency resource deployed (Here is a Resource model)

That why we see an error in log here: https://github.com/localstack/localstack/issues/3888#issuecomment-826131782

Actually this error will not effect the deploy result (all the resources still be deployed). And this fix will not effect the current logic code. It just make sure we don't have any exception during deployment.